### ISSUE TITLE
Xamarin Form App Crash using ZXingBarcodeImageView

### ISSUE THREAD

**Issue by Jeeepz** [Redth/ZXing.Net.Mobile#554]
I experience this error on android. So I have a tabbedpage as a NavigationPage. When I use navigation.PushAsync(somepage) and it shows the barcode, then when I hit the back button or click back at the toolbar, the app crashes. If I comment out the barcode, it works fine. Any solutions? 

Here's the code.
Going to the tabbedpage.
```
Tabs tabs = new Tabs();
NavigationPage.SetHasNavigationBar(tabs, false);
NavigationPage np = new NavigationPage(tabs);
App.Current.MainPage = np;
```
Going to another page.
 `Navigation.PushAsync(new CardPage((Card)e.Item));`

Here is the Barcode.
```
barcode = new ZXingBarcodeImageView
            {
                HorizontalOptions = LayoutOptions.FillAndExpand,
                VerticalOptions = LayoutOptions.FillAndExpand,
                AutomationId = "zxingBarcodeImageView",
                BarcodeFormat = ZXing.BarcodeFormat.EAN_13,
                BarcodeOptions =
                {
                    Width = 400,
                    Height = 75,
                    Margin = 10
                },
                BarcodeValue = card.CardNo  
            };
```
Then when I go back whole thing crashes...

**Comment by fredrikhaglund** [comment]
Try to rollback to nuget package version [2.1.47](https://www.nuget.org/packages/ZXing.Net.Mobile/2.1.47) and see if the problem also exists there or not. 

A huge refactoring was released just a couple weeks ago and new issues and some regression was introduced.

**Comment by abrari** [comment]
I'm rolling back to 2.1.12 and the problem is still there (crashing after hitting back button).

**Comment by boletus151** [comment]
Any update about this?

**Comment by manish-1988** [comment]
Same issue I am facing

**Comment by nikolaynn2008** [comment]
Hello, I need to save the result generated using
<forms:ZXingBarcodeImageView VerticalOptions="FillAndExpand" HorizontalOptions="FillAndExpand" BarcodeFormat="{Binding ValidatedSelectedFormat}" BarcodeValue="{Binding ValidatedCode}" Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="2" >
in jpeg format on the device. I searched for a lot of time on the Internet, I did not find the result. Can you help?
Thanks in advance,

**Comment by Char0394** [comment]
I fixed this issue by setting a default "FallbackValue"
ex. BarcodeValue="{Binding QRValue, FallbackValue='0'}"

**Comment by Redth** [comment]
This thing was buggy and is now fixed in the 3.0 series.

**Comment by GatoReno** [comment]
> I fixed this issue by setting a default "FallbackValue" ex. BarcodeValue="{Binding QRValue, FallbackValue='0'}"

this worked great for me
